File: src/commands/harness/status.ts
import type { Command } from "commander";
import { notImplemented } from "../../lib/errors/cliError.js";

export class HarnessStatusCommand {
  // Shows the status, event log, and result of one harness run.

  register(parent: Command): void {
    // Attaches `status <runId>` under the `vidbyte harness` group.
    parent
      .command("status <runId>")
      .description("Show the status and events of a harness run")
      .action(async (runId: string) => this.execute(runId));
  }

  private async execute(runId: string): Promise<void> {
    // Will fetch the run from the backend and render its status block.
    throw notImplemented("'vidbyte harness status'");
  }
}

File: src/vidbyte_cli/commands/harness/status.py
"""FILE: src/vidbyte_cli/commands/harness/status.py

PURPOSE: Fetches and renders one generic harness run snapshot.

TESTS: No feature tests are added under the approved no-tests workflow.
"""

from __future__ import annotations

import click

from ...lib.output import OutputDocument
from ...lib.runtime.context import ApplicationContext


class HarnessStatusCommand:
    def register(self, parent: click.Group) -> None:
        @parent.command(name="status", help="Show the status and events of a harness run")
        @click.argument("run_id")
        @click.pass_obj
        def _run(context: ApplicationContext, run_id: str) -> None:
            self.execute(context, run_id)

    def execute(self, context: ApplicationContext, run_id: str) -> None:
        run = context.harness_endpoints().get_run(run_id)
        context.output().result(
            OutputDocument(
                kind="harness.run",
                data={
                    "run_id": run.run_id,
                    "harness": run.harness,
                    "command": run.command,
                    "status": run.status,
                    "created_at": run.created_at,
                    "updated_at": run.updated_at,
                },
            ),
            context.harness_context().render.render_status(run),
        )
